backends/firrtl/firrtl.cc

```diff
@@ -392,7 +392,37 @@ struct FirrtlWorker
 		return result;
 	}
 
-	void run()
+	void emit_extmodule()
+	{
+		std::string moduleFileinfo = getFileinfo(module);
+		f << stringf("  extmodule %s: %s\n", make_id(module->name), moduleFileinfo.c_str());
+		vector<string> port_decls;
+
+		for (auto wire : module->wires())
+		{
+			const auto wireName = make_id(wire->name);
+			std::string wireFileinfo = getFileinfo(wire);
+
+			// Maybe not needed?
+			if (wire->port_id)
+			{
+				if (wire->port_input && wire->port_output)
+				{
+					log_error("Module port %s.%s is inout!\n", log_id(module), log_id(wire));
+				}
+				port_decls.push_back(stringf("    %s %s: UInt<%d> %s\n", wire->port_input ? "input" : "output",
+						wireName, wire->width, wireFileinfo.c_str()));
+			}
+		}
+
+		for (auto str : port_decls) {
```

Comment:
This should probably be:
```suggestion
		for (auto &str : port_decls) {
```
But I don't think it really makes a performance difference in this case.